src/openapi/streaming/connection/transport/signalr-core-transport.js

```diff
@@ -0,0 +1,337 @@
+        log.warn(
+            LOG_AREA,
+            'connection already exist, close the exisiting conection before starting new one',
+        );
+        return;
+    }
+
+    const protocol =
+        options.messageSerializationProtocol ||
+        new window.signalrCore.JsonHubProtocol();
+
+    try {
+        this.connection = buildConnection({
+            baseUrl: this.baseUrl,
+            contextId: this.contextId,
+            authToken: this.authToken,
+            protocol,
+        });
+    } catch (error) {
+        this.transportFailCallback(error);
+    }
+
+    this.connection.onclose = this.handleConnectionClosure.bind(this);
+
+    this.stateChangedCallback(constants.CONNECTION_STATE_CONNECTING);
+
+    return this.connection
+        .start()
+        .then(() => {
+            if (this.isStopping) {
+                return;
+            }
+
+            const messageStream = this.connection.stream('StartStreaming');
+            messageStream.subscribe({
+                next: (message) => this.handleNextMessage(message, protocol),
+                error: (error) => this.handleMessageStreamError(error),
+            });
+
+            this.messageStream = messageStream;
+
+            if (onStartCallback) {
+                onStartCallback();
+            }
+
+            this.stateChangedCallback(constants.CONNECTION_STATE_CONNECTED);
+        })
+        .catch((error) => {
+            this.transportFailCallback(error);
+        });
+};
+
+SignalrCoreTransport.prototype.stop = function() {
+    if (!this.connection) {
+        log.warn(LOG_AREA, "connection doesn't exist");
+        return;
+    }
+
+    this.isStopping = true;
+
+    // close message stream before closing connection
+    if (this.messageStream) {
+        return this.messageStream
+            .cancelCallback()
+            .then(() => this.connection.stop());
+    }
+
+    return this.connection.stop();
+};
+
+SignalrCoreTransport.prototype.handleConnectionClosure = function(error) {
+    if (error) {
+        log.error(LOG_AREA, 'connection closed abruptly', { error });
+    }
+
+    this.connection = null;
+    this.isStopping = false;
+
+    if (!this.hasStreamingStarted) {
+        this.transportFailCallback(error);
+        return;
+    }
+
+    this.hasStreamingStarted = false;
+    this.stateChangedCallback(constants.CONNECTION_STATE_DISCONNECTED);
+};
+
+SignalrCoreTransport.prototype.handleNextMessage = function(message, protocol) {
+    if (!this.hasStreamingStarted) {
+        this.hasStreamingStarted = true;
+    }
+
+    try {
+        const normalizedMessage = normalizeMessage(message, protocol);
+        const parsedMessage = parseMessage(normalizedMessage, this.utf8Decoder);
+
+        this.receivedCallback(parsedMessage);
+    } catch (error) {
+        log.error(LOG_AREA, 'Error occurred while parsing message', {
+            error,
+            payload: error.payload,
+        });
+
+        this.transportFailCallback();

```

Comment:
I double checked and we still fallback to other transport
we don't fallback in case we get parse error during protobuf message parsing 
thats what I saw in your earlier PR

Reply:
I see ok thats fine.